### PR TITLE
Migrate pallet-glutton benchmark to v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/glutton/src/benchmarking.rs [hunk 1]
```diff
@@ -20,80 +20,121 @@
 //! Has to be compiled and run twice to calibrate on new hardware.
 
 #[cfg(feature = "runtime-benchmarks")]
-use super::*;
-
-use frame_benchmarking::benchmarks;
+use frame_benchmarking::{v2::*, BenchmarkError};
```
Comment (re-gius) [inline substrate/frame/glutton/src/benchmarking.rs:23]:
```suggestion
use frame_benchmarking::v2::*;
```

`BenchmarkError` is already imported in `v2`

File: substrate/frame/glutton/src/benchmarking.rs [hunk 2]
```diff
@@ -20,80 +20,121 @@
...
+			Pallet::<T>::on_idle(
+				frame_system::Pallet::<T>::block_number(),
+				Weight::from_parts(WEIGHT_REF_TIME_PER_MILLIS * 100, WEIGHT_PROOF_SIZE_PER_KB * 20),
+			);
+		}
 	}
 
-	empty_on_idle {
-	}: {
-		// Enough weight to do nothing.
```
Comment (re-gius) [inline substrate/frame/glutton/src/benchmarking.rs:88]: I would keep the old comment here

**Review by re-gius — APPROVED** [review]

File: prdoc/pr_6296.prdoc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+title: Migrate pallet-glutton benchmark to v2
+doc:
+- audience: Runtime Dev
+  description: |-
+    Part of:
+
+    - #6202.
```
Comment (seadanda) [inline prdoc/pr_6296.prdoc:7]:
```suggestion
    Update `pallet-glutton` to benchmarks v2.
```

File: substrate/frame/glutton/src/benchmarking.rs [hunk 4]
```diff
@@ -20,80 +20,121 @@
...
-		Glutton::<T>::on_idle(System::<T>::block_number(), T::WeightInfo::empty_on_idle());
+	#[benchmark]
+	fn empty_on_idle() {
```
Comment (seadanda) [inline substrate/frame/glutton/src/benchmarking.rs:113]:
```suggestion
	fn empty_on_idle() {
		// Enough weight to do nothing.
```

**Review by seadanda — APPROVED** [review]
LGTM modulo these very minor comments. I can resolve and merge to avoid re-review

**Pull request merged**